pkg/export/export.go

```diff
@@ -475,12 +485,74 @@ func (e *Exporter) triggerNext() {
 	}
 }
 
-// ClientName and Version are used to identify to User Agent. TODO(maxamin): automate versioning.
 const (
+	// ClientName is used to identify the User Agent.
 	ClientName = "prometheus-engine-export"
-	Version    = "0.6.3"
+	// mainModuleVersion is the version of the main module. Align with git tag.
+	// TODO(TheSpiritXIII): Remove with https://github.com/golang/go/issues/50603
+	mainModuleVersion = "v0.7.0-rc.0"
+	// mainModuleName is the name of the main module. Align with go.mod.
+	mainModuleName = "github.com/GoogleCloudPlatform/prometheus-engine"
 )
 
+// Testing returns true if running within a unit test.
+// TODO(TheSpiritXIII): Replace with https://github.com/golang/go/issues/52600
+func Testing() bool {
+	return flag.Lookup("test.v") != nil
+}
+
+// Version is used in the User Agent. This version is automatically detected if
+// this function is imported as a library. However, the version is statically
+// set if this function is used in a binary in prometheus-engine due to Golang
+// restrictions. While testing, the static version is validated for correctness.
+func Version() (string, error) {
+	if Testing() {
+		// TODO(TheSpiritXIII): After https://github.com/golang/go/issues/50603 just return empty string here.
+		cmd := exec.Command("git", "describe", "--tags", "--abbrev=0")
+		var stdout, stderr bytes.Buffer
+		cmd.Stdout = &stdout
+		cmd.Stderr = &stderr
+
+		version := ""
+		if err := cmd.Run(); err != nil {
```

Comment:
Can you add a comment describing what this is doing? e.g. `// Prefer version from GIT_TAG env var, otherwise fallback to closest git tag`